xtext/editorserver/src/controllers/XtextController.js

```diff
@@ -57,15 +65,23 @@ class XtextController {
 
             // Read the build log
             try {
-                buildLog = fs.readFileSync(buildLogPath, 'utf8');
+                if (fs.existsSync(buildLogPath)) {
+                    buildLog = fs.readFileSync(buildLogPath, 'utf8');
+                } else {
+                    buildLog = "";
+                }
             } catch (err) {
                 console.log("Error reading build log: " + buildLogPath);
                 console.log(err);
             }
 
             // Read the build status
             try {
-                buildStatus = Number( fs.readFileSync(buildStatusPath, 'utf8') );
+                if (fs.existsSync(buildStatusPath)) {
+                    buildStatus = Number( fs.readFileSync(buildStatusPath, 'utf8') );
+                } else {
+                    buildStatus = 0;
```

Comment:
Setting buildStatus to 0 could be misleading as it indicates the build was successful. However, it is not provided in the response so just a minor point. Leaving buildStatus undefined the `buildStatus > 0`  check for error is always false. But could be made more explicit by setting it to null and adding a condition to check rather than implicitly relying on undefined.

Reply:
This isn't a problem because we now use explicit `editorDeployed` flag to check if process has finished.